src/Tempest.php

```diff
@@ -17,60 +16,60 @@
 {
     private function __construct(
         private Kernel $kernel,
-        private AppConfig $appConfig,
     ) {
     }
 
-    public static function boot(string $root, ?Closure $createAppConfig = null): self
+    public static function boot(string $root): self
     {
         $dotenv = Dotenv::createUnsafeImmutable($root);
+
         $dotenv->safeLoad();
 
-        $createAppConfig ??= fn () => new AppConfig(
-            environment: Environment::from(env('ENVIRONMENT', Environment::LOCAL->value)),
-            discoveryCache: env('DISCOVERY_CACHE', false),
-            enableExceptionHandling: env('EXCEPTION_HANDLING', false),
+        $kernel = new Kernel(
```

Comment:
The createAppConfig closure was only needed if we wanted to inject appconfig from the outside because dotenv had to load first. Without it, we can simply construct it